ui/src/app/workflows/components/workflow-dag/graph/shifter.ts

```diff
@@ -0,0 +1,39 @@
+// This shifter is used to shift the index of the first relevant item one index to the right, and subsequent one index to the left.
```

Comment:
I'm not sure what this class is meant to do?

Reply:
In short: the collapsed node always gets ordered before the other children: `C 1 2`, `C` is the collapsed node and `1, 2` are child nodes. What we want is `1 C 2`. Since we must "stream" the ordering due to the way the graph code works at the moment, this class serves to shift the index of `C` and `1` when so desired. It will essentially map `0 -> 1, 1 -> 0, 2 -> 2` for ordering purposes.

This is also helpful if there are multiple collapsed nodes in the same level: `Ca 1a 2a Cb 1b 2c -> 1a Ca 2a 1b Cb 2b`.

I can explain further if this is not clear

Reply:
please make commentary more detailed

Reply:
Added